src/bootstrap/lib.rs

```diff
@@ -1646,6 +1646,20 @@ fn chmod(path: &Path, perms: u32) {
 #[cfg(windows)]
 fn chmod(_path: &Path, _perms: u32) {}
 
+/// If code is not 0 (successful exit status), exit status is 101 (rust's default error code.)
+/// If the test is running and code is an error code, it will cause a panic.
+pub fn detail_exit(code: i32) -> ! {
+    // Successful exit
+    if code == 0 {
```

Comment:
Is this case ever hit? I don't think we should be using this function for successful exit, there's some cleanup in main.rs that should run. Can you instead assert the code isn't 0?

Reply:
No, for example, in src/bootstrap/flags.rs , exist code.
```zsh
% rg "exit\([^1-9]" *.rs                                                                  12:34:33
flags.rs
263:                crate::detail_exit(exit_code);
349:            crate::detail_exit(exit_code);

lib.rs
1651:fn detail_exit(code: i32) -> ! {
1654:        std::process::exit(0);

% rg "exit_code" *.rs      
flags.rs
262:                let exit_code = if args.is_empty() { 0 } else { 1 };
263:                crate::detail_exit(exit_code);
331:        let usage = |exit_code: i32, opts: &Options, verbose: bool, subcommand_help: &str| -> ! {
349:            crate::detail_exit(exit_code);
```
However, such code is not much(There are 2).
Would it be better to just change the flag.rs and insert assert in the exit function?

Reply:
Hmm, ok - seems fine to just leave it as you currently have it then.